core/src/test/scala/com/mesosphere/usi/core/SchedulerLogicHandlerTest.scala

```diff
@@ -0,0 +1,16 @@
+package com.mesosphere.usi.core
+
+import com.mesosphere.mesos.client.MesosCalls
+import com.mesosphere.usi.core.helpers.MesosMock
+import com.mesosphere.usi.core.models.{Goal, PodId, PodSpec, PodSpecUpdated, RunSpec}
+import com.mesosphere.utils.UnitTest
+
+class SchedulerLogicHandlerTest extends UnitTest {
+  "output a mesos call with revive for new podspec role" in {
```

Comment:
I would like to see some more unit tests here.

* Additional pod spec definitions for the same role results In additional revive calls
* An incoming offer which matches outstanding pendingLaunched pods results in a suppressed call